dbms/src/Functions/divide.cpp

```diff
@@ -62,7 +62,10 @@ struct TiDBDivideFloatingImpl<A, B, false>
     template <typename Result = ResultType>
     static Result apply(A a, B b)
     {
-        return static_cast<Result>(a) / b;
+        if constexpr (std::is_integral_v<Result>)
+            return (static_cast<Result>(a) + b / 2) / b;
```

Comment:
if `Result == UInt32`, $a = 2^{32} - 1$ and $b = 2^{32} - 1$, it may cause overflow?

Reply:
This does not seem to happen because for `Decimal32 = Decimal<Int32>`, the maxDecimalPrecision is 9. It means that a or b must be less than 999999999

Reply:
get it, thanks

Reply:
Now that I have some concerns about this, can you add some cases to make sure it doesn't overflow? /cc @LittleFall

Reply:
What happens when a is Int32::MAX?

Reply:
Add if it never overflows, please add some comments to explain why.

Reply:
changed implementation and added test.